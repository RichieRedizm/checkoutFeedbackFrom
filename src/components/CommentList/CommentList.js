import React from "react";
import Comment from "../Comment/Comment";

function CommentList(props) {
  return (
    <div className="comment-list">
      {props.comments.map((comment, index) => (
        <Comment key={index} comment={comment} />
      ))}
    </div>
  );
}

export default CommentList
