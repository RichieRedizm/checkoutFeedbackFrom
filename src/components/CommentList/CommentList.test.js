import React from 'react'
import Renderer from 'react-test-renderer'
import CommentList from './CommentList'
import Data from '../data'


describe('CommentList', () => {
  it('Should render a list of comments with default mock data', () => {

		const component = Renderer.create(<CommentList {...Data} />)
		const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})

