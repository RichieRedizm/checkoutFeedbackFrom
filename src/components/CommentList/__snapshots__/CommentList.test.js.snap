// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList Should render a list of comments with default mock data 1`] = `
<div
  className="comment-list"
>
  <div
    className="comment"
  >
    <h4
      className="name field"
    >
      Joe Blogs
    </h4>
    <span>
      joeblogs@gmail.com
    </span>
    <span
      className="rating"
    >
      2
    </span>
    <span>
      Example message
    </span>
  </div>
  <div
    className="comment"
  >
    <h4
      className="name field"
    >
      Richard Oakey
    </h4>
    <span>
      richieredizm@gmail.com
    </span>
    <span
      className="rating"
    >
      1
    </span>
    <span>
      Another example message
    </span>
  </div>
  <div
    className="comment"
  >
    <h4
      className="name field"
    >
      Testy McTester
    </h4>
    <span>
      testing@example.com
    </span>
    <span
      className="rating"
    >
      5
    </span>
    <span>
      And we have provided another example message
    </span>
  </div>
</div>
`;
